File: 06-python/problem_set/sentimental-mario-more/mario.py
import sys
from cs50 import get_int

# initialize n
n = 0

# get height of pyramid from user, validate user input
while n < 1 or n > 8:
    n = get_int("Height: ")

# print n rows, decrementing from n to 0
for row in range(n, 0, -1):
    # print spaces
    print(f" " * (row - 1), end="")
    # print hashes
    print("#" * (n - row + 1), end="")
    # print gap
    print(f"  ", end="")
    # print hashes and new line
    print("#" * (n - row + 1), end="")
    print()

File: 01-c/problem_set/mario-less/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
// get user input for pyramid height
    int n;
    do
    {
        n = get_int("Height of pyramid (range = 1-8 inclusive): ");
    }
    while (n < 1 || n > 8);

// print pyramid
    // for each row
    for (int i = 0; i < n; i++)
    {
        // print a decrementing number of spaces
        for (int spaces = n - 2; spaces >= i; spaces--)
        {
            printf(" ");
        }
        // print an incrementing number of #s
        for (int j = 0; j <= i; j++)
        {
            printf("#");
        }
        // print a newline at the end of the 'row' for loop
        printf("\n");
    }
}